src/Caching/Hybrid/src/Internal/DefaultHybridCache.StampedeKey.cs

```diff
@@ -0,0 +1,39 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Diagnostics.CodeAnalysis;
+
+namespace Microsoft.Extensions.Caching.Hybrid.Internal;
+
+partial class DefaultHybridCache
+{
+    internal readonly struct StampedeKey : IEquatable<StampedeKey>
+    {
+        private readonly string _key;
+        private readonly HybridCacheEntryFlags _flags;
```

Comment:
At the L1/L2 levels: no; but for the actual in-flight operations, we need to know which logic to apply; it would be very confusing if we joined in ways which meant they were completely ignored. I don't actively expect real situations that use different rules for the same key, but this is the best way to keep a clear idea of what should happen.

Reply:
I think you're saying that, if the flags differed, the requests wouldn't be part of the same stampede so we put the requests in different buckets?

Reply:
I'll add a comment, but: yes

Reply:
https://github.com/dotnet/aspnetcore/pull/55251